packages/flutter/lib/src/widgets/scroll_behavior.dart

```diff
@@ -225,7 +225,7 @@ class OverscrollBehavior extends BoundedBehavior {
 /// A scroll behavior that lets the user scroll beyond the scroll bounds only when the bounds are disjoint.
 class OverscrollWhenScrollableBehavior extends OverscrollBehavior {
   /// Creates a scroll behavior that allows overscrolling only when some amount of scrolling is already possible.
-  OverscrollWhenScrollableBehavior({ double contentExtent: 0.0, double containerExtent: 0.0, double minScrollOffset: 0.0 })
+  OverscrollWhenScrollableBehavior({ double contentExtent: double.INFINITY, double containerExtent: 0.0, double minScrollOffset: 0.0 })
```

Comment:
Per our discussion: this change should be applied to all of the scroll behaviors, since ScrollConfiguration produces OverscrollWhenScrollableBehavior by default but not unconditionally.

For the benefit of someone reading this code years from now, it might be good to explain the rationale for the double.INFINITY. We don't want to cause a scrollable's initial scroll offset, to be inadvertently clipped before a valid contentExtent is computed.